File: 2540_MinimumCommonValue/minimumCommonValue.js
// Time: o(n + m)
// space: o(1)
var getCommon = function(nums1, nums2) {
    let i = 0
    let j = 0

    while (i < nums1.length && j < nums2.length) {
    	if (nums1[i] === nums2[j]) {
    		return nums1[i]
    	} else if (nums1[i] < nums2[j]) {
    		i++
    	} else {
    		j++
    	}
    }
    return -1
};

console.log(getCommon([1, 2, 3], [4, 5, 6]))
console.log(getCommon([1, 2, 3], [2, 4]))
console.log(getCommon([1, 2, 3, 6], [2, 3, 4, 5]))

// there is a binary search way of doing it also. try that if you come across this problem

File: 2540_MinimumCommonValue/minimumCommonValue.py
# Time: o(n + m)
# space: o(1)
def getCommon(nums1: List[int], nums2: List[int]) -> int:
    i = j = 0

    while i < len(nums1) and j < len(nums2):
        if nums1[i] == nums2[j]:
            return nums1[i]
        elif nums1[i] < nums2[j]:
            i += 1
        else:
            j += 1

    return -1

print(getCommon([1, 2, 3], [4, 5, 6]))
print(getCommon([1, 2, 3], [2, 4]))
print(getCommon([1, 2, 3, 6], [2, 3, 4, 5]))

# there is a binary search way of doing it also. try that if you come across this problem